pkg/catalog/remote_catalog.go

```diff
@@ -77,28 +75,42 @@ func (c *RemoteCatalog) CreateOrUpdate(
 
 	diffApply, diffDelete := c.CalculateDiffs(moduleTemplatesRuntime, moduleTemplatesControlPlane)
 
+	results := make(chan error, len(diffApply)+len(diffDelete))
 	for _, diff := range diffApply {
-		diff.SetLastSync()
-
-		var buf bytes.Buffer
-
-		if err := json.NewEncoder(&buf).Encode(diff); err != nil {
-			return err
-		}
-		patch := client.RawPatch(types.ApplyPatchType, buf.Bytes())
+		diff := diff
+		go func() {
+			diff.SetLastSync()
+			if err := syncContext.RuntimeClient.Patch(
+				ctx, diff, client.Apply, c.settings.SSAPatchOptions,
+			); err != nil {
+				results <- fmt.Errorf("could not apply module template diff: %w", err)
+			} else {
+				results <- nil
+			}
+		}()
+	}
+	for _, diff := range diffDelete {
+		diff := diff
+		go func() {
+			if err := syncContext.RuntimeClient.Delete(ctx, diff); err != nil {
+				results <- fmt.Errorf("could not delete module template from diff: %w", err)
+			} else {
+				results <- nil
```

Comment:
I suggest we can use [errgroup](https://bostonc.dev/blog/go-errgroup) to deal with the concurrent collection of errors